### PR TITLE
Break aspnet Dockerfile templates into finer-grained sub-templates

### PR DESCRIPTION
**Pull request by mthalman**
These changes define a set of fine-grained aspnet Dockerfile sub-templates that enables more sharing of content than the previous templates. This has allowed all the templates to be distilled down primarily to a small set of templates pivoting on two characteristics: OS type and single-stage/multi-stage.

The primary Dockerfile templates are `eng/dockerfile-templates/aspnet/Dockerfile.linux` and `eng/dockerfile-templates/aspnet/Dockerfile.windows`. From there, it breaks off into a sub-template for either single-stage or multi-stage implementation. There are even further sub-templates for handling common content between them all such as installing ASP.NET and defining the environment variables.

### CODE REVIEW

**Comment by mthalman** [comment]
I realized I used operator functions instead of operator symbols so I'm going to fix that up to make things cleaner to read.

File: eng/dockerfile-templates/runtime/Dockerfile.linux.install-aspnet [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{{
```
Comment (MichaelSimons) [inline eng/dockerfile-templates/runtime/Dockerfile.linux.install-aspnet:1]: Is this file referenced?  Its existence doesn't seem right.  It is an aspnet template in the runtime directory.  Same goes for the next file (windows version).

File: eng/dockerfile-templates/aspnet/Dockerfile.linux.multi-stage [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{{
```
Comment (MichaelSimons) [inline eng/dockerfile-templates/aspnet/Dockerfile.linux.multi-stage:1]: Did you consider combining the multi-stage and single-stage into a single template?  Obviously there are tradeoffs.  The value is seeing both patterns together to help ensure consistency and that the two final stages do not diverge.  To be clear, I don't see value in trying to integrate the two inline rather combining the two templates into an if else but sharing the common variables.

File: eng/dockerfile-templates/runtime/Dockerfile.linux.install-aspnet (hunk 1 above)
Comment (mthalman) [reply eng/dockerfile-templates/runtime/Dockerfile.linux.install-aspnet]: Deleted them.

**Review by MichaelSimons — APPROVED** [review]
It's really nice to see so many duplicated templates removed/reduced.

**Pull request merged**